cli/cmd/cluster.go

```diff
@@ -56,96 +69,166 @@ type (
+			return nil, errors.New("you need Linkerd to be installed in order to setup a remote cluster")
+		}
+		return nil, err
+	}
 
-  # Display latency and health status about the remote gateways
-  linkerd --context=cluster-b cluster gateways`,
+	defaults, err := mccharts.NewValues()
+	if err != nil {
+		return nil, err
 	}
 
-	createCredentialsCommand := &cobra.Command{
+	defaults.GatewayName = opts.gatewayName
+	defaults.GatewayNamespace = opts.gatewayNamespace
+	defaults.IdentityTrustDomain = global.Global.IdentityContext.TrustDomain
+	defaults.IncomingPort = opts.incomingPort
+	defaults.LinkerdNamespace = controlPlaneNamespace
+	defaults.ProbePath = opts.probePath
+	defaults.ProbePeriodSeconds = opts.probePeriodSeconds
+	defaults.ProbePort = opts.probePort
+	defaults.ProxyOutboundPort = global.Proxy.OutboundPort.Port
+	defaults.ServiceAccountName = opts.serviceAccountName
+	defaults.ServiceAccountNamespace = opts.serviceAccountNamespace
+	defaults.NginxImageVersion = opts.nginxImageVersion
+	defaults.NginxImage = opts.nginxImage
+	defaults.LinkerdVersion = version.Version
+
+	return defaults, nil
+}
+
+func newGatewaysCommand() *cobra.Command {
+
+	opts := gatewaysOptions{}
+
+	cmd := &cobra.Command{
 		Hidden: false,
-		Use:    "create-credentials",
-		Short:  "Create the necessary credentials for service mirroring",
+		Use:    "gateways",
+		Short:  "Display stats information about the remote gateways",
 		Args:   cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
-
-			labels := map[string]string{
-				k8s.ControllerComponentLabel: k8s.ServiceMirrorLabel,
-				k8s.ControllerNSLabel:        controlPlaneNamespace,
+			req := &pb.GatewaysRequest{
+				RemoteClusterName: opts.clusterName,
+				GatewayNamespace:  opts.gatewayNamespace,
+				TimeWindow:        opts.timeWindow,
 			}
 
-			clusterRole := v1.ClusterRole{
-				ObjectMeta: metav1.ObjectMeta{Name: createOpts.serviceAccount, Namespace: createOpts.namespace, Labels: labels},
-				TypeMeta:   metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
-				Rules: []rbacv1.PolicyRule{
-					{
-						APIGroups: []string{""},
-						Resources: []string{"services"},
-						Verbs:     []string{"list", "get", "watch"},
-					},
-				},
+			client := checkPublicAPIClientOrExit()
+			resp, err := requestGatewaysFromAPI(client, req)
+			if err != nil {
+				return err
 			}
+			renderGateways(resp.GetOk().GatewaysTable.Rows, stdout)
+			return nil
+		},
+	}
 
-			svcAccount := corev1.ServiceAccount{
-				ObjectMeta: metav1.ObjectMeta{Name: createOpts.serviceAccount, Namespace: createOpts.namespace, Labels: labels},
-				TypeMeta:   metav1.TypeMeta{Kind: v1.ServiceAccountKind, APIVersion: "v1"},
-			}
+	cmd.Flags().StringVar(&opts.clusterName, "cluster-name", "", "the name of the remote cluster")
+	cmd.Flags().StringVar(&opts.gatewayNamespace, "gateway-namespace", "", "the namespace in which the gateway resides on the remote cluster")
```

Comment:
Do you mind also setting the defaults here to match the other ones?

Reply:
Addressed in #4307